### PR TITLE
add handle attribute for projections

### PR DESCRIPTION
**Pull request by DavidBadura**

### CODE REVIEW

File: tests/Unit/Projection/DefaultProjectionRepositoryTest.php [hunk 1]
```diff
@@ -128,6 +131,77 @@ public function drop(): void
...
+            Email::fromString('profile@test.com')
+        );
+
+        $projectionRepository = new DefaultProjectionRepository([$projection]);
+        $projectionRepository->handle($profileCreated);
+
+        self::assertSame($profileCreated, $projection::$handledEvent);
+
+        $profileVisited = ProfileVisited::raise(ProfileId::fromString('1'), ProfileId::fromString('2'));
+        $projectionRepository->handle($profileVisited);
```
Comment (DanielBadura) [inline tests/Unit/Projection/DefaultProjectionRepositoryTest.php]: The assert if the method got handled is missing

File: src/Projection/AttributeHandleMethod.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+use Patchlevel\EventSourcing\Aggregate\AggregateChanged;
+use Patchlevel\EventSourcing\Attribute\Handle;
+use ReflectionClass;
+
+use function array_key_exists;
+
+/**
+ * @psalm-require-implements Projection
+ */
+trait AttributeHandleMethod
```
Comment (DanielBadura) [inline src/Projection/AttributeHandleMethod.php:16]: we have a lot of traits atm tbh 😨

Comment (DavidBadura) [reply src/Projection/AttributeHandleMethod.php:16]: :man_shrugging:

Comment (DavidBadura) [reply src/Projection/AttributeHandleMethod.php:16]: Do you have a better idea? I think that in the next major release we can force attribute support and thereby remove all traits. But would still wait and collect ideas.

File: tests/Unit/Projection/DefaultProjectionRepositoryTest.php (hunk 1 above)
Comment (DavidBadura) [reply tests/Unit/Projection/DefaultProjectionRepositoryTest.php]: i rewrite the tests :)

File: src/Projection/AttributeHandleMethod.php (hunk 2 above)
Comment (DanielBadura) [reply src/Projection/AttributeHandleMethod.php:16]: No, no other idea atm

**Review by DanielBadura — APPROVED** [review]

**Pull request merged**